packages/core/src/lib/actions/env.ts

```diff
@@ -0,0 +1,54 @@
+import type { IncomingHttpHeaders } from "http"
+import type { AuthAction, AuthConfig } from "../../types"
+
+/**
+ * Set default env variables on the config object
+ **/
+export function setEnvDefaults(envObject: any, config: AuthConfig) {
+  try {
+    const url = process.env.AUTH_URL
+    if (url && !config.basePath) config.basePath = new URL(url).pathname
+  } catch {
+  } finally {
+    config.basePath ??= `/auth`
+  }
+
+  config.redirectProxyUrl ??= process.env.AUTH_REDIRECT_PROXY_URL
+  config.secret ??= process.env.AUTH_SECRET
```

Comment:
Could you incorporate https://github.com/nextauthjs/next-auth/blob/1345a8cfbc484aeda80bbc22fd753003b824f5fc/packages/next-auth/src/lib/env.ts#L14-L22 which was added in #9804?